src/context/loading-page-context.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, ReactNode, useState } from "react";
+
+interface ILoadingPage {
+  image: string;
+  title: string;
+  text: string;
+  isLoadingBarVisible: boolean;
+  isButtonVisible: boolean;
+  buttonText?: string;
+  buttonFunction?: () => void;
+}
+
+interface ILoadingPageContext {
```

Comment:
why do we need a context here ? cannot we just pass down props from the main component to the children ?
may you please give some hints about why we need another page to show loading progress, and cannot we just show a plain react component on loading ?